rcl/test/rcl/test_info_by_topic.cpp

```diff
@@ -0,0 +1,378 @@
+    free(temp_info_array->info_array);
+  });
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  const auto ret = rcl_get_publishers_info_by_topic(this->node_ptr,
+      &allocator, this->topic_name, false, temp_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestInfoByTopicFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_subscriptions_info_by_topic_invalid_participants)
+{
+  // temp_info_array is invalid because it is expected to be zero initialized
+  // and the info_array variable inside it is expected to be null.
+  const auto & temp_info_array = this->topic_info_array.get();
+  temp_info_array->info_array =
+    reinterpret_cast<rmw_topic_info_t *>(calloc(1, sizeof(rmw_topic_info_t)));
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+    free(temp_info_array->info_array);
+  });
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  const auto ret = rcl_get_subscriptions_info_by_topic(this->node_ptr,
+      &allocator, this->topic_name, false, temp_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+TEST_F(CLASSNAME(TestInfoByTopicFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_publishers_subscription_info_by_topic)
+{
+  // This is implemented only in fastrtps currently.
+  if (!is_fastrtps) {
+    GTEST_SKIP();
+  }
+  rmw_qos_profile_t default_qos_profile;
+  default_qos_profile.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+  default_qos_profile.depth = 0;
+  default_qos_profile.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+  default_qos_profile.durability = RMW_QOS_POLICY_DURABILITY_VOLATILE;
+  default_qos_profile.lifespan = {10, 0};
+  default_qos_profile.deadline = {11, 0};
+  default_qos_profile.liveliness_lease_duration = {20, 0};
+  default_qos_profile.liveliness = RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC;
+
+  rcl_ret_t ret;
+  const rosidl_message_type_support_t * ts = ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, Strings);
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+
+  rcl_publisher_t publisher = rcl_get_zero_initialized_publisher();
+  rcl_publisher_options_t publisher_options = rcl_publisher_get_default_options();
+  publisher_options.qos = default_qos_profile;
+  ret = rcl_publisher_init(
+    &publisher,
+    this->node_ptr,
+    ts,
+    this->topic_name,
+    &publisher_options);
+  ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+
+  rcl_subscription_t subscription = rcl_get_zero_initialized_subscription();
+  rcl_subscription_options_t subscription_options = rcl_subscription_get_default_options();
+  subscription_options.qos = default_qos_profile;
+  ret = rcl_subscription_init(
+    &subscription,
+    this->node_ptr,
+    ts,
+    this->topic_name,
+    &subscription_options);
+  ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  std::string fqdn = this->topic_name;
+  fqdn = "/" + fqdn;
```

Comment:
@jaisontj nit: can be collapsed into the line above and made `const`

Reply:
Done.